native-platform/src/main/cpp/win.cpp

```diff
@@ -88,16 +97,30 @@ DWORD get_file_stat(wchar_t* pathStr, jboolean followLink, file_stat_t* pFileSta
         }
         return error;
     }
-    pFileStat->lastModified = lastModifiedNanos(&attr.ftLastWriteTime);
-    if (attr.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) {
-        pFileStat->size = 0;
-        pFileStat->fileType = FILE_TYPE_DIRECTORY;
-    } else {
-        pFileStat->size = ((LONG64) attr.nFileSizeHigh << 32) | attr.nFileSizeLow;
-        pFileStat->fileType = FILE_TYPE_FILE;
-    }
+#ifdef WINDOWS_MIN
+    // Done, no Symlinks
+    fillFileStat(
+        pFileStat,
+        false,
+        attr.dwFileAttributes,
+        &attr.ftLastWriteTime,
+        attr.nFileSizeHigh,
+        attr.nFileSizeLow);
     return ERROR_SUCCESS;
-#else    //WINDOWS_MIN: Windows Vista+ support for symlinks
+#else //WINDOWS_MIN: Windows Vista+ support for symlinks
+    if ((attr.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) != FILE_ATTRIBUTE_REPARSE_POINT) {
+        // Done, not a symlink
+        fillFileStat(
+            pFileStat,
+            false,
+            attr.dwFileAttributes,
+            &attr.ftLastWriteTime,
+            attr.nFileSizeHigh,
+            attr.nFileSizeLow);
+        return ERROR_SUCCESS;
+    }
+
+    // Now let's try to follow the symlink or find out if the current reparse point is a symlink
     DWORD dwFlagsAndAttributes = FILE_FLAG_BACKUP_SEMANTICS;
     if (!followLink) {
```

Comment:
We need `CreateFileW` even if we don't want to follow the symlink to find out if it is a symlink, since we need a file handle for `GetFileInformationByHandleEx`. At this point we only know that the current file is a reparse point, which may or may not be a symlink.